src/scripting/lua_widget_attributes.cpp

```diff
@@ -0,0 +1,527 @@
+
+
+/// CLASSIC
+
+WIDGET_GETTER("value_compat,selected_index", int, gui2::listbox)
+{
+	return w.get_selected_row() + 1;
+}
+
+/* idea
+WIDGET_GETTER("selected_widget", gui2::widget, gui2::listbox)
+{
+	if(w.get_selected_row() >= w.get_item_count()) {
+		throw std::invalid_argument("widget has no selected item");
+	}
+	return w.get_row_grid(w.get_selected_row());
+}
+*/
+
+WIDGET_SETTER("value_compat,selected_index", int, gui2::listbox)
+{
+	w.select_row(value - 1);
+}
+
+WIDGET_GETTER("value_compat,selected_index", int, gui2::multi_page)
+{
+	return w.get_selected_page() + 1;
+}
+
+WIDGET_SETTER("value_compat,selected_index", int, gui2::multi_page)
+{
+	w.select_page(value -1);
+}
+
+WIDGET_GETTER("value_compat,selected_index", int, gui2::stacked_widget)
+{
+	return w.current_layer() + 1;
+}
+
+WIDGET_SETTER("value_compat,selected_index", int, gui2::stacked_widget)
+{
+	w.select_layer(value - 1);
+}
+
+WIDGET_GETTER("selected_index", int, gui2::selectable_item)
+{
+	return w.get_value() + 1;
+}
+
+WIDGET_SETTER("selected_index", int, gui2::selectable_item)
+{
+	if(value > int(w.num_states())) {
+		throw std::invalid_argument("invalid index");
+	}
+	w.set_value(value + 1);
+}
+
+WIDGET_GETTER("value_compat,selected", bool, gui2::selectable_item)
+{
+	if(w.num_states() == 2) {
+		return w.get_value_bool();
+	}
+	throw std::invalid_argument("invalid widget");
+}
+
+WIDGET_SETTER("value_compat,selected", bool, gui2::selectable_item)
+{
+	w.set_value_bool(value);
+}
+
+WIDGET_GETTER("value_compat,text", std::string, gui2::text_box)
+{
+	return w.get_value();
+}
+
+WIDGET_SETTER("value_compat,text", std::string, gui2::text_box)
+{
+	w.set_value(value);
+}
+
+WIDGET_GETTER("value_compat,value", int, gui2::slider)
+{
+	return w.get_value();
+}
+
+WIDGET_SETTER("value_compat,value", int, gui2::slider)
+{
+	w.set_value(value);
+}
+
+WIDGET_GETTER("value_compat,percentage", int, gui2::progress_bar)
+{
+	return w.get_percentage();
+}
+
+WIDGET_SETTER("value_compat,percentage", int, gui2::progress_bar)
+{
+	w.set_percentage(value);
+}
+
+WIDGET_GETTER("value_compat,selected_item_path", std::vector<int>, gui2::tree_view)
+{
+	auto res = w.selected_item()->describe_path();
+	for(int& a : res) { ++a;}
+	return res;
+}
+
+WIDGET_GETTER("path", std::vector<int>, gui2::tree_view_node)
+{
+	auto res = w.describe_path();
+	for(int& a : res) { ++a;}
+	return res;
+}
+
+WIDGET_SETTER("value_compat,unfolded", bool, gui2::tree_view_node)
+{
+	if(value) {
+		w.unfold();
+	} else {
+		w.fold();
+	}
+}
+
+WIDGET_SETTER("value_compat,unit", lua_index_raw, gui2::unit_preview_pane)
+{
+	if(const unit_type* ut = luaW_tounittype(L, value.index)) {
+		w.set_displayed_type(*ut);
+	} else if(unit* u = luaW_tounit(L, value.index)) {
+		w.set_displayed_unit(*u);
+	} else {
+		luaW_type_error(L, value.index, "unit or unit type");
+	}
+}
+
+WIDGET_GETTER("item_count", int, gui2::multi_page)
```

Comment:
I think the `#` operator should also be overloaded to return this value. You can probably just make the operator work for all widgets, returning `item_count` if it exists or 1 otherwise.

Reply:
hmm yes i actually had the same thought, just an hour ago